### ISSUE TITLE
Wavenet training doesnot work

### ISSUE THREAD

**Issue by lfzhai** [Rayhane-mamah/Tacotron-2#179]
Hi @Rayhane-mamah,  I'm trying the latest commit with tenserflow 1.10 on LJSpeech-1.1. The tacobon training seems no problem but wavenet training gives meaningless result. Synthesised wavs can not be understandable after 1300000 steps. And it is strange that step-XX-pred.wav and step-XX-real.wav are 7k size (too small for audio). The only change to haparam is reduced batch size. Expect to your reply. Thanks!

**Comment by tdplaza** [comment]
me too, and even in my case, it results negative loss value.

**Comment by b04901014** [comment]
@tdplaza I think negative loss is reasonable since WaveNet uses cross entropy as the loss function.

**Comment by b04901014** [comment]
@lfzhai I think negative loss is reasonable since it uses the negative cross entropy as the loss function.

**Comment by antontc** [comment]
I have the same problem.

**Comment by lkfo415579** [comment]
I think synthesize.py has some problems during wavenet vocoding. . .

**Comment by peter05010402** [comment]
@lkfo415579  and find another problem: if follow the hparams.py, when WaveNet training, an error will happen;  if change 
        hop_size = 300, #For 22050Hz, 275 ~= 12.5 ms
        win_size = 1200, #For 22050Hz, 1100 ~= 50 ms (If None, win_size = n_fft)
        sample_rate = 24000,
WaveNet training will be OK.
I do not know why?!

**Comment by lkfo415579** [comment]
@peter05010402  what do you mean.. ? you gave the default setting, it should be able to train the WaveNet model. (I am still getting noises from synthesizing mode)

**Comment by sysuzyx** [comment]
@lkfo415579 Could you tell me how long you trained for 800k wavenet model? I trained wavenet more than a week, and it only reached 160k steps on LJspeech with default hparams. The slow training is strange. The result is noise in logs-WaveNet/eval-dir/wavs and good audio in logs-WaveNet/wavs(is it GTA?)

The other question is: the sample rate of LJspeech is 22050Hz and the default setting is _sample_rate = 24000_. Why model can be trained though the sample rate is not matched? Is it upsampling in the preprocessing?

**Comment by mertcokluk** [comment]
I did not manage to launch training. Can you tell me the issue right away, did I miss something? I used  preprocess:'python wavenet_preprocess.py', training:   'python train.py --model=WaveNet'

And I got the following:
InvalidArgumentError (see above for traceback): Conv2DCustomBackpropInputOp only supports NHWC.

[[{{node WaveNet_model/inference/conv_transpose1d/conv2d_transpose}} = Conv2DBackpropInput[T=DT_FLOAT, data_format="NCHW", dilations=[1, 1, 1, 1], padding="SAME", strides=[1, 1, 1, 5], use_cudnn_on_gpu=true, _device="/job:localhost/replica:0/task:0/device:CPU:0"](WaveNet_model/inference/conv_transpose1d/stack, WaveNet_model/inference/conv_transpose1d/kernel/read, WaveNet_model/inference/ExpandDims_1)]]

**Comment by tdplaza** [comment]
> @lkfo415579 Could you tell me how long you trained for 800k wavenet model? I trained wavenet more than a week, and it only reached 160k steps on LJspeech with default hparams. The slow training is strange. The result is noise in logs-WaveNet/eval-dir/wavs and good audio in logs-WaveNet/wavs(is it GTA?)
> 
> The other question is: the sample rate of LJspeech is 22050Hz and the default setting is _sample_rate = 24000_. Why model can be trained though the sample rate is not matched? Is it upsampling in the preprocessing?

yes, it does. 

The librosa.load() function (in load_wav() function) works like this: if you pass sample_rate, it automatically resample your signal to match that sample rate. If you do not pass sample rate, It will use the default value as 22050Hz. If you want to keep the original rate, you have to pass sample_rate=None.

**Comment by tdplaza** [comment]
> I did not manage to launch training. Can you tell me the issue right away, did I miss something? I used preprocess:'python wavenet_preprocess.py', training: 'python train.py --model=WaveNet'
> 
> And I got the following:
> InvalidArgumentError (see above for traceback): Conv2DCustomBackpropInputOp only supports NHWC.
> 
> [[{{node WaveNet_model/inference/conv_transpose1d/conv2d_transpose}} = Conv2DBackpropInput[T=DT_FLOAT, data_format="NCHW", dilations=[1, 1, 1, 1], padding="SAME", strides=[1, 1, 1, 5], use_cudnn_on_gpu=true, _device="/job:localhost/replica:0/task:0/device:CPU:0"](WaveNet_model/inference/conv_transpose1d/stack, WaveNet_model/inference/conv_transpose1d/kernel/read, WaveNet_model/inference/ExpandDims_1)]]

You are having problem with tensorflow-gpu. 

If you are using GPU to train, make sure that you have totally remove tensorflow package and install tensorflow-gpu. Do not install both of them.

**Comment by lfzhai** [comment]
Has anyone trained wavenet model and got good samples by now? I've trained 1500000 steps and still much noise with the syntheized audio, any tips?
